src/EditorFeatures/Core/Extensibility/Navigation/NavigableItemFactory.cs

```diff
@@ -59,6 +59,16 @@ private static IEnumerable<Location> GetPreferredSourceLocations(ISymbol symbol)
                 : locations.Where(loc => loc.IsInSource);
         }
 
+        public static IEnumerable<INavigableItem> GetPreferredNavigableItems(Solution solution, IEnumerable<INavigableItem> navigableItems)
+        {
+            var generatedCodeRecognitionService = solution.Workspace.Services.GetService<IGeneratedCodeRecognitionService>();
+            navigableItems = navigableItems.Where(n => n.Document != null);
+            var hasNonGeneratedCodeItem = navigableItems.Any(n => !generatedCodeRecognitionService.IsGeneratedCode(n.Document));
+            return hasNonGeneratedCodeItem ?
+                navigableItems.Where(n => !generatedCodeRecognitionService.IsGeneratedCode(n.Document)) :
+                navigableItems.Where(n => generatedCodeRecognitionService.IsGeneratedCode(n.Document));
```

Comment:
I have retained the same generated code based filtering that was already being done for GotoDefinition - I think we at least want to be consistent in both code paths. Can you please open a separate issue for your recommended change?